server/game/cards/01_SOR/events/AsteroidSanctuary.ts

```diff
@@ -0,0 +1,33 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { UnitCard } from '../../../core/card/CardTypes';
+import { EventCard } from '../../../core/card/EventCard';
+import { RelativePlayer, WildcardLocation } from '../../../core/Constants';
+
+export default class AsteroidSanctuary extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '6901817734',
+            internalName: 'asteroid-sanctuary'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Exhaust an enemy unit and give a shield to a friendly unit that costs 3 or less',
+            targetResolvers: {
+                exhaust: {
+                    locationFilter: WildcardLocation.AnyArena,
+                    controller: RelativePlayer.Opponent,
+                    immediateEffect: AbilityHelper.immediateEffects.exhaust()
+                },
+                shield: {
+                    controller: RelativePlayer.Self,
+                    cardCondition: (card) => (card as UnitCard).cost <= 3,
```

Comment:
We have a typescript typing trick available that makes this a little simpler (and the type conversion safer):

```typescript
cardCondition: (card) => card.isUnitCard() && card.cost <= 3,
```

The `card.isUnitCard()` will cause `card` to be identified as type `UnitCard` if the check succeeds. Won't go into the technical details unless you're interested, but that's our preferred way of handling this situation now (might be some older code that still does it the other way).

Reply:
Ahh, this is a nice use of type predicates! Done in https://github.com/SWU-Karabast/forceteki/pull/88/commits/2b1010e504f7710ed6260379ea9f7e47893d0901 

Also, fixed the doc where I found the original usage.